project/tmp54/README.md

```diff
@@ -0,0 +1,63 @@
+# R-Tree Search Engine
+
+## Problem to Solve
+
+R-Tree is a tree data structure used to represent a set of object in an n-dimensional space. Each node of the tree is the Minimum Bounding Rectangle (MBR) of the child nodes. The leaf node is an object in the n-dimensional space.
+
+## System Architecture
+
+- Builder pattern
+- Fluent API
```

Comment:
This is not exactly architecture, but I like it to be mentioned in a proposal.  You can probably include it in the API section.